### PR TITLE
removed problems with empty centers, added json file for ease of use

### PR DESCRIPTION
**Pull request by UKnowWhoIm**
#### Bug Description
- Some districts like Kottayam(304) have no centers, so payload would be `{"centers": []}` 
- Thus `checkAvailability` to return `False, False` instead of string as expected, 
- So `if (len(available) > 0)` would throw `TypeError`

#### Bug Fix
- Change `if len(available) > 0` to `if available`

#### Feature
- A `settings.json` can be provided as an alternative of passing 5 shell arguments.
- If settings file is detected, settings would be loaded from that file, else shell arguments would be used.
- A `settings.example.json` file is provided with required schema.

### CODE REVIEW

**Review by rohitanil — COMMENTED** [review]
Could you update the README file as well about the settings.json file usage. 
Let's keep the settings.json file and not settings.example.json (as it might confuse non techie people) and have a detailed readme on how arguments can be passed via settings file.

**Comment by UKnowWhoIm** [comment]
What if we entirely depend on `settings.json`, ie no command line arguments?

Then an entrypoint script can be created for docker, which would make it it even more easier to run

**Comment by rohitanil** [comment]
> What if we entirely depend on `settings.json`, ie no command line arguments?
> 
> Then an entrypoint script can be created for docker, which would make it it even more easier to run



> What if we entirely depend on `settings.json`, ie no command line arguments?
> 
> Then an entrypoint script can be created for docker, which would make it it even more easier to run

Yea. That's the idea. Use ENTRYPOINT in Dockerfile and let the user put in the arguments in a json file. The only concern is to just keep settings.json and not the sample json file. 

Also test it locally

**Comment by UKnowWhoIm** [comment]
### Changes
- Moved `settings.json` to `settings/settings.json` and volume mounted settings to container to persist settings.
- Added `initial_setup.py` which will check settings.json for errors, and ask the users to input data if it finds any

**Review by rohitanil — COMMENTED** [review]
Do we want yo make it this complex with all the additional docker commands? Lets keep a settings.json file where users can add their arguments before they build the docker image. And within the Dockerfile, copy the json file and in the main script read from it. 
Maybe I'm missing something.

File: README.md [hunk 1]
```diff
@@ -19,10 +19,15 @@ Once all the above requirements are met, do the following
 2. Open Terminal and change your directory to the folder you have just cloned/ downloaded. (Remember to extract the .zip file if you have downloaded it)
 3. Make sure Docker Desktop is running.
 4. Run `docker build -t covin --rm .`
-5. Run `docker run -it --name covin-schedule --rm covin`
-6. Run `python covin_slot_tracker.py <DISTRICT_ID> <TWILIO AUTH TOKEN> <TWILIO ACCOUNT SID> <TWILIO PHONE NUMBER> <YOUR PHONE NUMBER>` in the interactive shell. Make sure your phone number has the country code. eg: +919xxxxxxxx
+5. Run `docker run -it --mount 'type=volume,src=settings,dst=/home/app_user/settings' --name covin-schedule --rm covin`
+6. Enter your settings
```
Comment (rohitanil) [inline README.md]: Will there be a prompt as in what value we need to enter?

**Review by rohitanil — COMMENTED** [review]
The idea is to make it less technical as possible. Minimum user interactions

File: README.md (hunk 1 above)
Comment (UKnowWhoIm) [reply README.md]: Yep

**Comment by UKnowWhoIm** [comment]
> Yea. That's the idea. Use ENTRYPOINT in Dockerfile and let the user put in the arguments in a json file. The only concern is to just keep settings.json and not the sample json file.
>
I thought you meant user input via shell, that's why i had to add volume.

Letting the user modify settings.json is honestly a better and simpler way

**Comment by UKnowWhoIm** [comment]
I've made it simple, but settings.json exists in the git index, 
- If I remove it, git will delete the file in the next commit,
- If I don't remove it git will track changes of that file, ie User data could be leaked.

Solutions
1. Create a setup script which creates the local settings.json from user input, but without volume the change won't persist, ie every time it executes, user will be prompted for input.
To make it less complex, we could add a setup shell script to run the docker commands, end user won't know the difference.

2. Create a settings_example.json

**Comment by UKnowWhoIm** [comment]
@rohitanil

**Review by rohitanil — COMMENTED** [review]
@UKnowWhoIm looks good. Let me test it

**Comment by UKnowWhoIm** [comment]
What about `settings.json` stil being in the index @rohitanil? Anyone could accidentally commit their actual credentials to the settings file

**Pull request merged**